api/pkg/instrumentation/metrics/prometheus.go

```diff
@@ -85,3 +111,28 @@ func (p PrometheusClient) MeasureDurationMs(
 		p.MeasureDurationMsSince(key, starttime, labels)
 	}
 }
+
+// MeasureGauge takes in the Metric name, the count and a map of labels and values
+// to be associated to the metric. Error will be thrown if errors occur in accessing
+// the metric or associating the labels.
+func (p PrometheusClient) MeasureGauge(
```

Comment:
This method is ideally added to the `Collector` interface in `metrics.go` and implemented for `Nop` as well. 'Gauge' is a common enough metric type so its name should be ok to use in the interface too.

Since Gauge is not measuring something per se, can we call it `RecordGauge` instead of  `MeasureGauge` ?